paddle/pir/core/type_util.cc

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2023 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "paddle/pir/core/type_util.h"
+#include <algorithm>
+
+namespace pir {
+
+Type GetElementTypeOrSelf(Type type) {
+  if (auto sType = type.dyn_cast<ShapedTypeInterface>())
+    return sType.GetElementType();
+  return type;
+}
+
+bool VerifyCompatibleShape(phi::DDim shape1, phi::DDim shape2) {
```

Comment:
```suggestion
bool VerifyCompatibleShape(const phi::DDim& shape1, const phi::DDim& shape2) {
```
对于非 Impl 的数据结构在入参形式上，推荐使用const &，因为他们的copy 构造成本并不是想象中的那么低，尤其对于这类底层高频的方法

Reply:
Ok，一并修改